p2panda-rs/src/schema/validate/blob.rs

```diff
@@ -0,0 +1,151 @@
+// SPDX-License-Identifier: AGPL-&3.0-or-later
+
+use once_cell::sync::Lazy;
+use regex::Regex;
+
+use crate::operation::plain::{PlainFields, PlainValue};
+use crate::schema::validate::error::BlobError;
+
+/// @TODO: Do we want a max and what should it be?
+const MAX_BLOB_LENGTH: i64 = 100000; // 1GB
+
+/// Checks "length" field of operations with "blob_v1" schema id.
+///
+/// 1. It must be less than `MAX_BLOB_LENGTH`
+pub fn validate_length(value: &i64) -> bool {
+    *value <= MAX_BLOB_LENGTH
+}
+
+/// Checks "mime_type" field of operations with "blob_v1" schema id.
+///
+/// 1. It matches expected mime type format
+pub fn validate_mime_type(value: &str) -> bool {
```

Comment:
We could use this crate to test against existing mime types: https://docs.rs/mime/latest/mime/

Reply:
I tried this crate for mime type validation, but it didn't catch many invalid mime types (which my regex did catch :rofl: ). It hasn't been updated in 4 years so maybe it's just out of date...

Reply:
Did you try others?

Reply:
Not yet, there's this one which looks more up-to-date https://github.com/picoHz/mediatype

Reply:
Hmmm, tried that one too, it seems to not care too much if something is an existing mime type, so things like this:

`"this/mime.type.hasoneelementwhichiswaytoolong"`

parse into a valid media type.

Reply:
What behaviour are we expecting here? If we want to be strict we could hard code a list of known mime types and validate against this.

Reply:
Yeah I guess we shouldn't control that, let's stick to the regex 🖖🏻